### ISSUE TITLE
Export of data doesn't work after update to 1.10.0

### ISSUE THREAD

**Issue by saschafoerster** [pretix/pretix#705]
After updating to 1.10.0, I tested to create a checkin list as PDF. The browser shows: 

> Ein unerwarteter Fehler ist aufgetreten.

Logfiles: pretix.log
`ERROR 2017-12-10 22:39:54,482 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)`

Thanks! ;)

**Comment by saschafoerster** [comment]
Trying to buy a test-ticket, leads to the same error, but without trace in the logfiles. 
<img width="523" alt="bildschirmfoto 2017-12-11 um 00 04 49" src="https://user-images.githubusercontent.com/1153741/33810610-2e8c0e18-de07-11e7-8d76-3038ee0572e1.png">

**Comment by saschafoerster** [comment]
PS: The Progress indicator doesn't really look like in the preview (but maybe this is worth another issue report).

**Comment by raphaelm** [comment]
> PS: The Progress indicator doesn't really look like in the preview (but maybe this is worth another issue report).

For this one, can you try running ``docker exec -it pretix.service pretix upgrade`` again or changing and reverting the color of your shop? Your shop somehow has the old CSS.

**Comment by raphaelm** [comment]
If you didn't run ``docker exec -it pretix.service pretix upgrade`` at all, this might also be reason for the first problem. Restart the service afterwards, to be sure.

**Comment by saschafoerster** [comment]
I tried several times to upgrade and restart, with no change for export or CSS. Is there anything else I can do or give you more detail? Try for instance here: 
https://ticket.bonn.digital/bonndigital/vorlage/

**Comment by saschafoerster** [comment]
Maybe it depends of my faulty installation of this plugin? https://github.com/pretix/pretix-printtool

**Comment by saschafoerster** [comment]
Or maybe this is more helpful

> Last login: Mon Dec 11 00:37:58 on ttys000
MacBookMF:~ Ikarus84$ ssh bonn.digital
Linux sfvm01 4.9.0-4-amd64 #1 SMP Debian 4.9.51-1 (2017-09-28) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
root@sfvm01:~# cd /home/pretix
root@sfvm01:/home/pretix# docker pull pretix/standalone:stable
stable: Pulling from pretix/standalone
Digest: sha256:ecb9cd5d40ac1c8a57808e2e49b3a4853d25a71264d91ed2857f2f4f562efdf3
Status: Image is up to date for pretix/standalone:stable
root@sfvm01:/home/pretix# nano Dockerfile 
root@sfvm01:/home/pretix# docker build -t bonndigital/pretix .
Sending build context to Docker daemon  6.144kB
Step 1/9 : FROM pretix/standalone:stable
 ---> f0793b024bb7
Step 2/9 : USER root
 ---> Using cache
 ---> 8836085eb695
Step 3/9 : COPY id_pretix_enterprise /root/.ssh/id_rsa
 ---> Using cache
 ---> 5fd859aa895d
Step 4/9 : COPY id_pretix_enterprise.pub /root/.ssh/id_rsa.pub
 ---> Using cache
 ---> 02354a5f9a22
Step 5/9 : RUN chmod -R 0600 /root/.ssh
 ---> Using cache
 ---> 0bd44e870fc4
Step 6/9 : RUN pip3 install pretix-passbook pretix-sepadebit pretix-cartshare pretix-pages pretix-fontpack-free pretix-newsletter-ml pretix-wirecard
 ---> Using cache
 ---> 098ad12a325d
Step 7/9 : RUN mkdir -p /etc/ssh && ssh-keyscan -t rsa -p 10022 code.rami.io >> /root/.ssh/known_hosts && echo StrictHostKeyChecking=no >> /root/.ssh/config && pip3 install -Ue "git+ssh://git@code.rami.io:10022/pretix/pretix-slack.git@stable#egg=pretix-slack"
 ---> Using cache
 ---> 8ba5af2c901a
Step 8/9 : RUN cd /pretix/src && sudo -u pretixuser make production
 ---> Using cache
 ---> d6fb1d9ad457
Step 9/9 : USER pretixuser
 ---> Using cache
 ---> cbe69f7254d2
Successfully built cbe69f7254d2
Successfully tagged bonndigital/pretix:latest
root@sfvm01:/home/pretix# systemctl restart pretix.service
root@sfvm01:/home/pretix# docker exec -it pretix.service pretix upgrade
System check identified some issues:

WARNINGS:
?: (mysql.W002) MySQL Strict Mode is not set for database connection 'default'
	HINT: MySQL's Strict Mode fixes many data integrity problems in MySQL, such as data truncation upon insertion, by escalating warnings into errors. It is strongly recommended you activate it. See: https://docs.djangoproject.com/en/1.11/ref/databases/#mysql-sql-mode
Operations to perform:
  Apply all migrations: auth, banktransfer, contenttypes, easy_thumbnails, otp_static, otp_totp, paypal, pretix_cartshare, pretix_pages, pretix_sepadebit, pretixbase, pretixdroid, pretixmultidomain, sessions, stripe
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
root@sfvm01:/home/pretix# docker exec -it pretix.service pretix upgrade
System check identified some issues:

WARNINGS:
?: (mysql.W002) MySQL Strict Mode is not set for database connection 'default'
	HINT: MySQL's Strict Mode fixes many data integrity problems in MySQL, such as data truncation upon insertion, by escalating warnings into errors. It is strongly recommended you activate it. See: https://docs.djangoproject.com/en/1.11/ref/databases/#mysql-sql-mode
Operations to perform:
  Apply all migrations: auth, banktransfer, contenttypes, easy_thumbnails, otp_static, otp_totp, paypal, pretix_cartshare, pretix_pages, pretix_sepadebit, pretixbase, pretixdroid, pretixmultidomain, sessions, stripe
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
root@sfvm01:/home/pretix# less /var/pretix-data/logs/pretix.log
root@sfvm01:/home/pretix# nano Dockerfile 
root@sfvm01:/home/pretix# python -V
Python 2.7.13
root@sfvm01:/home/pretix# less /var/pretix-data/logs/
csp.log     pretix.log  
root@sfvm01:/home/pretix# less /var/pretix-data/logs/
csp.log     pretix.log  
root@sfvm01:/home/pretix# less /var/pretix-data/logs/csp.log 
root@sfvm01:/home/pretix# less /var/pretix-data/logs/csp.log 
root@sfvm01:/home/pretix# less /var/pretix-data/logs/pretix.log

ERROR 2017-12-12 21:38:12,710 celery.utils.dispatch.signal signal Signal handler <bound method DjangoFixup.on_import_modules of <celery.fixups.django.DjangoFixup object at 0x7f83f09754a8>> raised: PermissionError(13, 'Permission denied')
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 50, in <module>
    import reportlab_mods   #application specific modifications can be anywhere on python path
ModuleNotFoundError: No module named 'reportlab_mods'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/celery/utils/dispatch/signal.py", line 227, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 84, in on_import_modules
    self.worker_fixup.validate_models()
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 122, in validate_models
    self.django_setup()
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 118, in django_setup
    django.setup()
  File "/usr/local/lib/python3.6/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.6/site-packages/django/apps/registry.py", line 116, in populate
    app_config.ready()
  File "/pretix/src/pretix/base/__init__.py", line 13, in ready
    from . import invoice  # NOQA
  File "/pretix/src/pretix/base/invoice.py", line 11, in <module>
    from reportlab.lib import pagesizes
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 53, in <module>
    _fake_import(os.path.expanduser(os.path.join('~','.reportlab_mods')),'reportlab_mods')
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 27, in _fake_import
    sys.modules[name] = m.load_module(name)
  File "<frozen importlib._bootstrap_external>", line 399, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 823, in load_module
  File "<frozen importlib._bootstrap_external>", line 682, in load_module
  File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 674, in exec_module
  File "<frozen importlib._bootstrap_external>", line 780, in get_code
  File "<frozen importlib._bootstrap_external>", line 832, in get_data
PermissionError: [Errno 13] Permission denied: '/root/.reportlab_mods'
ERROR 2017-12-12 21:38:50,234 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)
ERROR 2017-12-12 21:43:53,483 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)

**Comment by saschafoerster** [comment]
`python -v` gives me "Python 2.7.13" on Debian 9. But maybe it's about the version in the docker container...

**Comment by raphaelm** [comment]
> PermissionError: [Errno 13] Permission denied: '/root/.reportlab_mods'

That might be it. There is a bug in the reportlab package that I already tried to work around but apparently it still happens. I'm unsure why I don't get the error, though. I pushed e9e15520 right now, the docker container should be built in an hour by docker hub.

Could you try building pulling pretix/standalone:latest then and building on top of that for a test instead of :stable?

**Comment by saschafoerster** [comment]
How shall I say it... maybe with a GIF, didn't work...
![bug](https://user-images.githubusercontent.com/1153741/33965897-d8be81ec-e05d-11e7-8f78-9d09284ca6e3.gif)

**Comment by saschafoerster** [comment]
Besides everything is really slow. It takes 30 seconds to load a page. Celery and python3 are taking up to 50-70% of CPU some times...

**Comment by raphaelm** [comment]
Since there now have been two reports of problems with the docker installation (#705 and #716), I created a shell script that closely follows the tutorial and executed it in a fresh virtual machine with debian stretch installed. I was thereby able to verify that the tutorial is correct and the resulting pretix instance was working for me. You can follow my steps and compare with your setup here: https://github.com/pretix/pretix-test-tutorials/blob/master/docker_smallscale/install.sh

Therefore, it appears to me that this problem is without the scope of the tutorial or the pretix software and somehow related to your server setup.

I found one bug during this procedure, the template ``/etc/systemd/system/pretix.service`` did only specify the docker container to be started as ``pretix/standalone``, not ``pretix/standalone:stable`` which *might* have caused that you run off the current git master instead of the stable version. Since I do not recommend downgrades, I suggest to stick with that until 1.11 comes out and then change this line to ``pretix/standalone:stable``. This shouldn't cause the problem reported here, though, since we also run the git master in production.

**Comment by saschafoerster** [comment]
I had to go back to pretix 1.9.1, there everything still works. How can that be? I'm still ready to help fund the bug. 🐜

**Comment by saschafoerster** [comment]
To be also more detailed: I compared all the config-files and options with the recommended options, and they are mostly exactly the same. I wiped all docker images and container, I tried fresh installations following the instructions with :stable as well as with  :master, each with fresh docker installations. Nothing did help. After hours of testing, I tried to go back to version 1.9.1 and here, everything works, even faster as before. 
- Is it possible, that some data in the SQL database is corrupted and that causes the failure? Maybe by installation of one of the new plugins, that was not needed? Maybe they had some migration script attached? I guessed this because of this error message: "pretix.base.async async"

Here are my lastest error messages (before downgrading)

> ERROR 2017-12-14 22:44:52,827 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)
ERROR 2017-12-14 23:02:24,431 celery.utils.dispatch.signal signal Signal handler <bound method DjangoFixup.on_import_modules of <celery.fixups.django.DjangoFixup object at 0x7f66f04976a0>> raised: PermissionError(13, 'Permission denied')
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 50, in <module>
    import reportlab_mods   #application specific modifications can be anywhere on python path
ModuleNotFoundError: No module named 'reportlab_mods'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/celery/utils/dispatch/signal.py", line 227, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 84, in on_import_modules
    self.worker_fixup.validate_models()
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 122, in validate_models
    self.django_setup()
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 118, in django_setup
    django.setup()
  File "/usr/local/lib/python3.6/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.6/site-packages/django/apps/registry.py", line 116, in populate
    app_config.ready()
  File "/pretix/src/pretix/base/__init__.py", line 13, in ready
    from . import invoice  # NOQA
  File "/pretix/src/pretix/base/invoice.py", line 11, in <module>
    from reportlab.lib import pagesizes
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 53, in <module>
    _fake_import(os.path.expanduser(os.path.join('~','.reportlab_mods')),'reportlab_mods')
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 27, in _fake_import
    sys.modules[name] = m.load_module(name)
  File "<frozen importlib._bootstrap_external>", line 399, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 823, in load_module
  File "<frozen importlib._bootstrap_external>", line 682, in load_module
  File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 674, in exec_module
  File "<frozen importlib._bootstrap_external>", line 780, in get_code
  File "<frozen importlib._bootstrap_external>", line 832, in get_data
PermissionError: [Errno 13] Permission denied: '/root/.reportlab_mods'
ERROR 2017-12-14 23:03:32,820 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)
ERROR 2017-12-14 23:04:08,183 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)
ERROR 2017-12-14 23:14:28,992 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)
ERROR 2017-12-14 23:19:26,841 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)

If it searches for " File "/usr/local/lib/python3.6" where does it search python? Locally or in the docker-container?

Is it possible to copy the database to a different place, try again, and copy it back? Or are there side-results? 

So far my ideas.

**Comment by saschafoerster** [comment]
Even with a fresh data base I get this error messages: 

> less /var/pretix-data/logs/pretix.log 

  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 411, in _query
    rowcount = self._do_query(q)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 374, in _do_query
    db.query(q)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/connections.py", line 277, in query
    _mysql.connection.query(self, query)
django.db.utils.ProgrammingError: (1146, "Table 'pretix.pretixbase_user' doesn't exist")
INFO 2017-12-15 00:01:26,394 celery.app.trace trace Task pretix.base.services.update_check.update_check[b7e6509c-5415-46b1-b89d-0f40321784ef] succeeded in 0.49259952000011253s: None
ERROR 2017-12-15 00:07:51,623 celery.utils.dispatch.signal signal Signal handler <bound method DjangoFixup.on_import_modules of <celery.fixups.django.DjangoFixup object at 0x7f519179f710>> raised: PermissionError(13, 'Permission denied')
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 50, in <module>
    import reportlab_mods   #application specific modifications can be anywhere on python path
ModuleNotFoundError: No module named 'reportlab_mods'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/celery/utils/dispatch/signal.py", line 227, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 84, in on_import_modules
    self.worker_fixup.validate_models()
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 122, in validate_models
    self.django_setup()
  File "/usr/local/lib/python3.6/site-packages/celery/fixups/django.py", line 118, in django_setup
    django.setup()
  File "/usr/local/lib/python3.6/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.6/site-packages/django/apps/registry.py", line 116, in populate
    app_config.ready()
  File "/pretix/src/pretix/base/__init__.py", line 13, in ready
    from . import invoice  # NOQA
  File "/pretix/src/pretix/base/invoice.py", line 11, in <module>
    from reportlab.lib import pagesizes
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 53, in <module>
    _fake_import(os.path.expanduser(os.path.join('~','.reportlab_mods')),'reportlab_mods')
  File "/usr/local/lib/python3.6/site-packages/reportlab/__init__.py", line 27, in _fake_import
    sys.modules[name] = m.load_module(name)
  File "<frozen importlib._bootstrap_external>", line 399, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 823, in load_module
  File "<frozen importlib._bootstrap_external>", line 682, in load_module
  File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 674, in exec_module
  File "<frozen importlib._bootstrap_external>", line 780, in get_code
  File "<frozen importlib._bootstrap_external>", line 832, in get_data
PermissionError: [Errno 13] Permission denied: '/root/.reportlab_mods'
ERROR 2017-12-15 00:08:43,791 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)
ERROR 2017-12-15 00:08:48,882 pretix.base.async async Unexpected exception: NotRegistered("'pretix.base.services.export.export'",)

Maybe this is helpful? 

> django.db.utils.ProgrammingError: (1146, "Table 'pretix.pretixbase_user' doesn't exist")

**Comment by saschafoerster** [comment]
Last but not least for today, but maybe also unrelated: 

> # docker exec -it pretix.service pretix upgrade
System check identified some issues:
WARNINGS:
?: (mysql.W002) MySQL Strict Mode is not set for database connection 'default'
	HINT: MySQL's Strict Mode fixes many data integrity problems in MySQL, such as data truncation upon insertion, by escalating warnings into errors. It is strongly recommended you activate it. See: https://docs.djangoproject.com/en/1.11/ref/databases/#mysql-sql-mode
Operations to perform:
  Apply all migrations: auth, banktransfer, contenttypes, easy_thumbnails, otp_static, otp_totp, paypal, pretix_pages, pretixbase, pretixdroid, pretixmultidomain, sessions, stripe
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them."

**Comment by jfwiebe** [comment]
I have the same css problem after upgrading the container, see #718.

**Comment by raphaelm** [comment]
I was able to reproduce the problem and just released 1.10.1 which should fix the problem. Make sure to read the notes on docker installations at the end of the blogpost:
https://pretix.eu/about/de/blog/20171216-release-1101/

**Comment by jfwiebe** [comment]
Awesome! Will try it out this evening.
JF

**Comment by saschafoerster** [comment]
Because I was working with a self-build docker-container (to get all the plugins I need, and sometimes even the ones, I don't really need), my line in the pretix.service-file was anyway:
>     bonndigital/pretix all
instead of 
>     pretix/standalone:stable all
I tried once to add stable to my line, but the build then failed. 

I changed the line in the dockerfile following the blogarticle to: 
> FROM pretix/standalone@sha256:04c15d3b668fe865d4a6adf0fcaa871a001de02128d25e084db2bc473c6846f9 
without "all" at the end. 

After the update and migration everything worked smooth and perfect! Thanks @raphaelm for your time, your service, your efforts for trying again and of course this wonderful tool.
